brouter-routing-app/src/main/java/btools/routingapp/BRouterActivity.java

```diff
@@ -730,4 +759,123 @@ public void onRequestPermissionsResult (int requestCode, String[] permissions, i
       }
     }
   }
+
+
+  private void import_profile(Intent intent) {
+    TextView scanResults;
+    //  app was activated by an intent with "action_view"...
+    System.out.println("Brouter Intent.ACTION_VIEW detected");
+
+    if (intent.getData() == null) {
+      System.out.println("Brouter intent.get(data) is NULL   ERROR");
+    }
+    else
+    {
+      setContentView(R.layout.import_intent);
+      scanResults = (TextView) findViewById(R.id.Info_brouter);
+      scanResults.setText("Start importing profile: \n");
+
+      Uri dataUri = intent.getData();
+      System.out.println("Brouter DATA=" + dataUri);
+
+// by some apps (bluemail) the file name must be "extracted" from the URI, as example with the following code
+// see https://stackoverflow.com/questions/14364091/retrieve-file-path-from-caught-downloadmanager-intent
+// URI example ==>  dat=content://me.bluemail.mail.attachmentprovider/a2939069-76b5-44e4-8cbd-94485d0fd4ff/cc32b61d-97a6-4871-b67f-945d1d1d43c8/VIEW
+      String filename = null;
+      Long filesize = null;
+      Cursor cursor = null;
+      try {
+        cursor = this.getContentResolver().query(intent.getData(), new String[]{
+          OpenableColumns.DISPLAY_NAME, OpenableColumns.SIZE}, null, null, null);
+        if (cursor != null && cursor.moveToFirst()) {
+          filename = cursor.getString(0);
+          filesize = cursor.getLong(1);
+        }
+      } finally {
+        if (cursor != null)
+          cursor.close();
+      }
+      System.out.println("Brouter filename=" + filename + "\n file size=" + filesize);
+      scanResults.setText(scanResults.getText() + "File name=" + filename + "\nFile size=" + filesize + " bytes\n");
+      //  is the file extention ".brf" in the file name
+      if (filename.indexOf(".brf") != -1 && (filename.indexOf(".brf") == (filename.length() - 4))) {
+        System.out.println("Brouter  OK, file extention is .brf!!!");
+      } else {
+        System.out.println("Brouter  ERROR, please provide a file with the valid extention \".brf\"...");
+        // report error in UI and stop
+        scanResults.setText(scanResults.getText() + "ERROR: File extention must be \".brf\" \n");
+        return;
+      }
+      // profile size is generally < 30 kb, so set max size to 100 kb
+      if (filesize > 100000) {
```

Comment:
The 100000 should be defined as static const in the class.

Reply:
Any enhancement is good